ci/steps/upload_test_results.py

```diff
@@ -0,0 +1,136 @@
+import dataclasses
+import json
+import os
+import sys
+from botocore.utils import merge_dicts
+
+import dacite
+import glci.model
+import glci.util
+import glci.s3
+
+
+def do_find(
+    architecture: str,
+    cicd_cfg_name: str,
+    committish: str,
+    gardenlinux_epoch: str,
+    modifiers: str,
+    platform: str,
+    repo_dir: str,
+    version: str,
+):
+    cicd_cfg = glci.util.cicd_cfg(cfg_name=cicd_cfg_name)
+    find_release = glci.util.preconfigured(
+        func=glci.util.find_release,
+        cicd_cfg=cicd_cfg,
+    )
+    modifiers = tuple(modifiers.split(','))
+    release = find_release(
+        release_identifier=glci.model.ReleaseIdentifier(
+            build_committish=committish,
+            version=version,
+            gardenlinux_epoch=int(gardenlinux_epoch),
+            architecture=glci.model.Architecture(architecture),
+            platform=platform,
+            modifiers=modifiers,
+        ),
+    )
+    return release
+
+
+def upload_test_results(
+    architecture: str,
+    cicd_cfg_name: str,
+    committish: str,
+    gardenlinux_epoch: str,
+    modifiers: str,
+    platform: str,
+    repo_dir: str,
+    version: str,
+):
+    if os.path.exists('/workspace/skip_tests'):
+        print('Tests already uploaded in previous run, skipping upload step')
+        sys.exit(0)
+
+    manifest = glci.model.ReleaseManifest(
+      build_committish=committish,
+      version=version,
+      build_timestamp=None,
+      gardenlinux_epoch=gardenlinux_epoch,
+      architecture=glci.model.Architecture(architecture).value,
+      platform=platform,
+      modifiers=modifiers,
+      paths=None,
+      published_image_metadata=None,
+    )
+
+    cicd_cfg = glci.util.cicd_cfg(cfg_name=cicd_cfg_name)
+    s3_client = glci.s3.s3_client(cicd_cfg)
+    aws_cfg_name = cicd_cfg.build.aws_cfg_name
+    s3_bucket_name = cicd_cfg.build.s3_bucket_name
+    modifiers = tuple(modifiers.split(','))
+
+    test_result_file_name = os.path.join(repo_dir, 'test_results.json')
+    if not os.path.exists(test_result_file_name):
+        print("No file found with test results, tests did not run, won't upload.")
```

Comment:
nit: consistent use of `'` or `"` for Strings. Not really necessary to change it here, but we should keep it in mind for the future.